src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingTxWithDoubleSpendSpecification_Steps.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Linq;
+using FluentAssertions;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Features.Wallet.Controllers;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.IntegrationTests.TestFramework;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Xunit.Abstractions;
+
+namespace Stratis.Bitcoin.IntegrationTests.Wallet
+{
+    public partial class SendingTxWithDoubleSpend : BddSpecification
+    {
+        private NodeBuilder builder;
+        private CoreNode stratisSender;
+        private CoreNode stratisReceiver;
+        private Transaction transaction;
+        private ErrorResult errorResult;
+        private HdAddress sendto;
+
+        public SendingTxWithDoubleSpend(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        protected override void BeforeTest()
+        {
+            this.builder = NodeBuilder.Create();
+            this.stratisSender = this.builder.CreateStratisPowNode();
+            this.stratisReceiver = this.builder.CreateStratisPowNode();
+
+            this.builder.StartAll();
+            this.stratisSender.NotInIBD();
+            this.stratisReceiver.NotInIBD();
+        }
+
+        protected override void AfterTest()
+        {
+            this.builder.Dispose();
+        }
+
+        private void wallets_with_coins()
+        {
+            var mnemonic1 = this.stratisSender.FullNode.WalletManager().CreateWallet("123456", "mywallet");
+            var mnemonic2 = this.stratisReceiver.FullNode.WalletManager().CreateWallet("123456", "mywallet");
+            mnemonic1.Words.Length.Should().Equals(12);
+            mnemonic2.Words.Length.Should().Equals(12);
+
+            var addr = this.stratisSender.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference("mywallet", "account 0"));
+            var wallet = this.stratisSender.FullNode.WalletManager().GetWalletByName("mywallet");
+            var key = wallet.GetExtendedPrivateKeyForAddress("123456", addr).PrivateKey;
+
+            this.stratisSender.SetDummyMinerSecret(new BitcoinSecret(key, this.stratisSender.FullNode.Network));
+            var maturity = (int)this.stratisSender.FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.stratisSender.GenerateStratis(maturity + 5);
```

Comment:
I seem to remember hearing we were suppsoed to use GenerateStratisWithMiner method rather than this

Reply:
Agreed - done